src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period "
+                + dateFrom + " - "
+                + dateTo);
+        LocalDate fromDate = LocalDate.parse(dateFrom, DATE_FORMATTER);
+        LocalDate toDate = LocalDate.parse(dateTo, DATE_FORMATTER);
+        int salary = 0;
+        for (String name : names) {
+            for (String dataString : data) {
+                String[] splitData = dataString.split(" ");
+                int hoursInData = Integer.parseInt(splitData[2]);
+                int incomePerHourInData = Integer.parseInt(splitData[3]);
```

Comment:
We perform parsing of data even if cases when it will not be required.
Please move inside if block where we will be sure these actions are necesssary